difftest/dpi_t1rocketemu/src/interconnect.rs

```diff
@@ -18,26 +30,64 @@ impl AddrInfo {
   }
 }
 
+/// Payload of a memory request
+#[derive(Debug)]
+pub enum MemReqPayload<'a> {
+  Read,
+  Write(&'a [u8], Option<&'a [bool]>),
+}
+
+/// A memory request
+///
+/// The device should keep the ordering of requests of the same ID
+#[derive(Debug)]
+pub struct MemReq<'a> {
+  pub id: u64,
+  pub payload: MemReqPayload<'a>,
+  pub addr: AddrInfo,
+}
+
+impl<'a> MemReqPayload<'a> {
+  fn is_write(&self) -> bool {
+    match self {
+      MemReqPayload::Read => false,
+      MemReqPayload::Write(_, _) => true,
+    }
+  }
+}
+
+/// Payload of memory response
+///
+/// Here we distinguish a fixed-sized MMIO register response
+/// from a arbitrarily-sized bulk memory response to simplify
+/// the implementation of RegDevice, since fixed-sized arrays
+/// are not borrowed
+#[derive(Debug)]
+pub enum MemRespPayload<'a> {
+  ReadBuffered(&'a [u8]),
+  ReadRegister([u8; 4]),
+  WriteAck,
+}
+
+/// A memory response
+///
+/// IDs correlates the response with the request
+#[derive(Debug)]
+pub struct MemResp<'a> {
+  pub id: u64,
+  pub payload: MemRespPayload<'a>,
+}
+
 // Caller is reponsible to ensure the following conditions hold:
 //   addr.len > 0
 //   addr.len == data.len()
 //   addr.len == mask.len() (if mask present)
 // However, since the functions are safe,
 // even if contracts violate, implementions must not break memory safety,
```

Comment:
These comments become outdated. Remove them or move them to `AddrInfo`/`MemReqPayload`

Reply:
Done in 68d735c3